Description: GNM: avoid potential issue with CPL circular buffers (can happen if there are extra drivers that use CPLGet....() functions)
Author: Even Rouault <even dot rouault at spatialys dot com>
Origin: https://trac.osgeo.org/gdal/changeset/41303/

--- a/gnm/gnm_frmts/file/gnmfilenetwork.cpp
+++ b/gnm/gnm_frmts/file/gnmfilenetwork.cpp
@@ -101,10 +101,10 @@ CPLErr GNMFileNetwork::Open(GDALOpenInfo
     }
 
     m_poLayerDriver = m_pMetadataDS->GetDriver();
-    const char* pszExt = CPLGetExtension(soMetadatafile);
+    CPLString osExt = CPLGetExtension(soMetadatafile);
 
     CPLString soGraphfile = CPLFormFilename(m_soNetworkFullName,
-                                            GNM_SYSLAYER_GRAPH, pszExt);
+                                            GNM_SYSLAYER_GRAPH, osExt);
     m_pGraphDS = (GDALDataset*) GDALOpenEx( soGraphfile, GDAL_OF_VECTOR |
                                             GDAL_OF_UPDATE, NULL, NULL, NULL );
     if( NULL == m_pGraphDS )
@@ -120,7 +120,7 @@ CPLErr GNMFileNetwork::Open(GDALOpenInfo
     }
 
     CPLString soFeaturesfile = CPLFormFilename(m_soNetworkFullName,
-                                            GNM_SYSLAYER_FEATURES, pszExt);
+                                            GNM_SYSLAYER_FEATURES, osExt);
     m_pFeaturesDS = (GDALDataset*) GDALOpenEx( soFeaturesfile, GDAL_OF_VECTOR |
                                                GDAL_OF_UPDATE, NULL, NULL, NULL );
     if( NULL == m_pFeaturesDS )
